rust/core-lib/src/find.rs

```diff
@@ -47,27 +53,72 @@ pub struct FindStatus {
     matches: usize
 }
 
+/// ID to uniquely identify a search query.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct FindId(pub(crate) usize);
```

Comment:
This is quite similar to `ViewId`. I am not sure if there is a more generic solution for this since some code is duplicated?

Reply:
`ViewId` is a string for historical reasons; you could consider having the find id just be a usize?